src/components/Files/FileItem.tsx

```diff
@@ -0,0 +1,129 @@
+}
+
+export const FileItem = ({ item, level, onEvent }: Props) => {
+  const selectedFilesIds = useAppSelector(
+    FilesSelectors.selectSelectedFilesIds,
+  );
+  const [isSelected, setIsSelected] = useState(
+    selectedFilesIds.includes(item.id),
+  );
+  const handleCancelFile = useCallback(() => {
+    onEvent?.('cancel', item.id);
+  }, [item.id, onEvent]);
+
+  const handleToggleFile = useCallback(() => {
+    setIsSelected((value) => !value);
+    onEvent?.('toggle', item.id);
+  }, [item.id, onEvent]);
+
+  const handleRetry = useCallback(() => {
+    onEvent?.('retry', item.id);
+  }, [item.id, onEvent]);
+
+  const handleRemove = useCallback(() => {
+    onEvent?.('remove', item.id);
+  }, [item.id, onEvent]);
+
+  return (
+    <div
+      className="group/file-item flex justify-between gap-3 px-3 py-1.5"
+      style={{
+        paddingLeft: (level && `${0.875 + level * 1.5}rem`) || '0.875rem',
+      }}
+    >
+      <div className="flex items-center gap-2 overflow-hidden">
+        <div className="group/file-icon text-gray-500">
+          {!isSelected && item.status !== 'FAILED' ? (
+            <IconFile className="group-hover/file-icon:hidden" size={18} />
+          ) : (
+            item.status === 'FAILED' && (
+              <IconExclamationCircle
+                className="shrink-0 text-red-800 dark:text-red-400"
+                size={18}
+              />
+            )
+          )}
+          <div
+            className={classNames(
+              'relative h-[18px] w-[18px] group-hover/file-icon:flex',
+              isSelected ? 'flex' : 'hidden',
+            )}
+          >
+            <input
+              className="checkbox-form peer h-[18px] w-[18px]"
+              type="checkbox"
+              checked={isSelected}
+              onChange={handleToggleFile}
+            />
+            <IconCheck
+              width={16}
+              height={16}
+              size={16}
+              className="pointer-events-none invisible absolute text-blue-500 peer-checked:visible"
+            />
+          </div>
+        </div>
+        <span
+          className={classNames(
+            'block max-w-full truncate',
+            item.status === 'FAILED' && 'text-red-800 dark:text-red-400',
+          )}
+        >
+          {item.name}
+        </span>
+      </div>
+
+      <div className="flex items-center">
+        {item.status === 'UPLOADING' && (
+          <div className="h-[3px] w-[60px] overflow-hidden rounded-full bg-gray-100 dark:bg-gray-700">
+            <div
+              className="h-full bg-blue-500"
+              style={{ width: `${item.percent}%` }}
+            ></div>
+          </div>
+        )}
+        {item.status === 'FAILED' && (
+          <button onClick={handleRetry}>
+            <IconReload className="shrink-0 text-gray-500" size={18} />
+          </button>
+        )}
+        {item.status && ['UPLOADING', 'FAILED'].includes(item.status) ? (
```

Comment:
`['UPLOADING', 'FAILED']` should be moved in contants outside component(new array each time) and wrapped in `new Set` 
```suggestion
        {item.status && statusesAllowCancel.has(item.status) ? (
```